src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/MethodBuilderImpl.cs

```diff
@@ -18,6 +17,8 @@ internal sealed class MethodBuilderImpl : MethodBuilder
         private readonly CallingConventions _callingConventions;
         private readonly TypeBuilderImpl _declaringType;
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
```

Comment:
Instead of adding these warnings everywhere, it would be better to change `GetTypeFromCoreAssembly` to take enum and have the warning supression on `GetTypeFromCoreAssembly` method only.

Reply:
It would be best to structure the code to be correct by construction and preserve the runtime types locally in case this is running against runtime reflection. Something like:
```csharp
private readonly static s_coreTypes = { typeof(void), typeof(object), ... };

[UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode", "Types are preserved via s_coreTypes"]
internal Type GetTypeFromCoreAssembly(CoreTypeId typeId)
{
    if (_coreTypes == null)
    {
        // Use s_coreTypes directly for runtime reflection
        _coreTypes = (_coreAssembly == typeof(object).Assembly) ? s_coreTypes : new Type[s_coreTypes.Length];
    }

    return _coreTypes[(int)typeId] ?? (_coreTypes[(int)typeId] = _coreAssembly.GetType(s_coreTypes[(int)typeId].FullName, throwOnError: true));
}
```

Reply:
Great, thank you!